src/main/resources/org/opentripplanner/apis/gtfs/schema.graphqls

```diff
@@ -2700,6 +2726,15 @@ type QueryType {
 
         """Only include routes, which use one of these modes"""
         transportModes: [Mode]
+
+        """
+        Only include routes whose pattern operates on at least one service date specified by this filter.
+
+        **Note**: A service date is a technical term useful for transit planning purposes and might not
+        correspond to a how a passenger thinks of a calendar date. For example, a night bus running
+        on Sunday morning at 1am to 3am, might have the previous Saturday's service date.
+        """
+        limitByPatternServiceDates: ServiceDateFilterInput
```

Comment:
This query starts to starts to have more and more arguments which don't necessarily interact too well with each other. However, I don't know if it makes sense to create a new query just for this as you would probably need some of the other arguments at some point at least.

Reply:
That is a good point. Lets discuss the options in the dev meeting.

Reply:
I deprecated the id parameter and explained what we should do instead.